src/ui/TreeView.js

```diff
@@ -28,46 +29,44 @@ class TreeView {
 
   update (json) {
     if (this.view) {
-      yo.update(this.view, this.render(json), {onBeforeElUpdated: (fromEl, toEl) => {
-        toEl.style.display = fromEl.style.display
-        toEl.className = fromEl.className
-        return true
-      }})
+      yo.update(this.view, this.render(json))
     }
   }
 
-  renderObject (item, parent, key, expand) {
+  renderObject (item, parent, key, expand, keyPath) {
     var data = this.extractData(item, parent, key)
     var children = (data.children || []).map((child, index) => {
-      return this.renderObject(child.value, data, child.key, expand)
+      return this.renderObject(child.value, data, child.key, expand, keyPath + '_' + child.key)
```

Comment:
`_` is a valid character for identifiers, so the assignment can be ambiguous. What about `','` or `' '`?